stock_storage_type/models/stock_location.py

```diff
@@ -312,9 +312,39 @@ def _domain_location_storage_type_constraints(
         )
         return pertinent_loc_storagetype_domain
 
+    def _allowed_locations_for_location_storage_types(
+        self, location_storage_types, quants, products
+    ):
+        valid_location_ids = set()
+        for loc_storage_type in location_storage_types:
+            location_domain = loc_storage_type._domain_location_storage_type(
+                self, quants, products
+            )
+            _logger.debug("pertinent location domain: %s", location_domain)
+            locations = self.search(location_domain)
+            valid_location_ids |= set(locations.ids)
+        return self.browse(valid_location_ids)
+
+    def _select_valid_locations(self, limit=None):
```

Comment:
"select valid locations"... for what? I think is not clear here. Maybe `_select_valid_putaway_locations`?